### DISCUSSION TITLE
What is the happy path for providing a custom expression to `sea_orm::query::Select`?

### DISCUSSION THREAD

**Discussion by emchristiansen** [SeaQL/sea-orm#1505] [Q&A] [answered]
What is the happy path for appending a custom expression to a query like this?

```rust
let results = MyTable::find()
  .all(db)
  .await?;
```

For example, I'd like to append `"AS OF TIMESTAMP TIDB_BOUNDED_STALENESS(NOW() - INTERVAL 60 SECOND, NOW())"` so that the query goes to a read replica and not the master in my TiDB cluster.

I can construct a `sea_query::query::SelectStatement` if I use `.into_query()` like this:

```rust
let something = MyTable::find()
    .into_query()
    .expr(Expr::val("AS OF TIMESTAMP TIDB_BOUNDED_STALENESS(NOW() - INTERVAL 60 SECOND, NOW())"))
```

But, now I have a `SelectStatement`, not a `Select<Entity>`, and I'm not sure how or whether I can go back to `Select<Entity>`.
I'm guessing I can just pipe it through raw SQL, but that seems like the wrong thing to do.

What's the right thing to do?

Thanks!

**Comment by tyt2y3** [comment, marked as answer]
I guess you are looking for https://docs.rs/sea-query/latest/sea_query/expr/struct.Expr.html#method.cust_with_expr ?

**Reply by emchristiansen** [reply]
Yup, that works.

I was struggling trying to figure out how to execute the resulting query, and finally stumbled upon this pattern:

```
// Created using MyTable::find().into_query()...
let select = ...

let statement = db.get_database_backend().build(&select);
let results = db.query_all(statement).await?
```

**Reply by tyt2y3** [reply]
Thank you for sharing
